services/management/grpc/annotation_server.go

```diff
@@ -54,9 +58,88 @@ func (as *AnnotationServer) AddAnnotation(ctx context.Context, req *managementpb
 		return nil, status.Error(codes.Unauthenticated, "Authorization error.")
 	}
 
-	_, err := as.grafanaClient.CreateAnnotation(ctx, req.Tags, time.Now(), req.Text, authorizationHeaders[0])
-	if err != nil {
-		return nil, err
+	var service, node bool
+	var serviceErr, nodeErr error
+	if len(req.ServiceName) > 0 {
+		service = true
+	}
+	if req.NodeName != "" {
+		node = true
+	}
+
+	if service {
+		serviceErr = findService(as.db, req.ServiceName)
```

Comment:
I guess it will be more simple: 

for serviceName in range req.ServiceNames {
    if  _, err = models.FindServiceByName(tx.Querier, sn); err != nil {
 	return  fmt.Errof("%s human readable error service with such name is not registered.", serviceName)
 }
}

and same for nodes.